src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java

```diff
@@ -0,0 +1,42 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="EmptyBlock">
+      <property name="tokens" value="LITERAL_CASE,LITERAL_DEFAULT"/>
+    </module>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.blocks.emptyblock;
+
+// xdoc section -- start
+public class Example3 {
+
+  private void test(int a) {
+
+    switch (a) {
+      case 1: someMethod();
+      default: // OK, as there is no block
+    }
+
+    switch (a) {
+      case 1 -> { } // violation, 'Must have at least one statement'
+      default -> { } // violation, 'Must have at least one statement'
+    }
+
+    switch (a) {
+      case 1 : { }  // violation, 'Must have at least one statement'
+      default : { }  // violation, 'Must have at least one statement'
+    }
```

Comment:
Let me play devils advocate here.

Suppose that you are a developer that wrote the second case here:
```java
    switch (a) {
      case 1 : { }  // violation, 'Must have at least one statement'
      default : { }  // violation, 'Must have at least one statement'
    }
```

You can "fix" these violations like this:
```java
    switch (a) {
      case 1 :
      default :
    }
```
Now suppose you wrote this:
```java
    switch (a) {
      case 1 -> { } // violation, 'Must have at least one statement'
      default -> { } // violation, 'Must have at least one statement'
    }
```

How would you fix this code?

Reply:
I would completely remove the case label. 
Let me play in reverse and consider this similar case.

```Java
        int x = 0;
        if (x == 1) {   // violation
          
        } else {  // violation

        }
```
How would you fix this code? you can't remove the `{}` but you will remove the if statement.

Reply:
@nrmancuso ping

Reply:
After playing around with this locally, I don’t think the compiler will let us have a case where we would be in the position to place a violation on code that would fail compilation.

I was concerned that we would place a violation on code, user would remove it, then the code would not compile. I do not think this is possible.

I am good here.